### ISSUE TITLE
[Feature Request] RefreshIndicator with non 0 index.

### ISSUE THREAD

**Issue by njovy** [peng8350/flutter_pulltorefresh#366]
```dart
import 'package:flutter/cupertino.dart';
import 'package:flutter/material.dart';

import 'package:pull_to_refresh/pull_to_refresh.dart';

class Test extends StatefulWidget {
  @override
  _TestState createState() => _TestState();
}

class _TestState extends State<Test> {
  RefreshController refreshController = RefreshController();

  @override
  Widget build(BuildContext context) {
    return CupertinoPageScaffold(
      child: SmartRefresher.builder(
        controller: refreshController,
        enablePullDown: true,
        builder: (context, physics) => buildCupertinoBody(context, physics),
        onRefresh: () {},
      ),
    );
  }

  Widget buildCupertinoBody(BuildContext context, ScrollPhysics physics) {
    return CustomScrollView(
      physics: physics,
      slivers: [
        CupertinoSliverNavigationBar(
          heroTag: "test",
          largeTitle: Text("TEST"),
        ),
        MaterialClassicHeader(
          offset: 50,
        ),
        SliverList(
          delegate: SliverChildBuilderDelegate(
            (BuildContext context, int index) {
              return Container(
                height: 150.0,
                child: Text("index=$index"),
              );
            },
            childCount: 10,
          ),
        )
      ],
    );
  }
}
```
Above is a minimal code that can have a non zero indexed ```RefreshIndicator```.

SmartRefresher only checks the firstChild for finding a RefreshIndicator. This behavior makes the ```RefreshIndicator``` scrolls all the way up and on top of the ```CupertinoSliverNavigationBar```.  I want to display the ```RefreshIndicator```  under the ```CupertinoSliverNavigationBar``` when users pull down.

**Comment by peng8350** [comment]
you can consider about using NestedScrollView (header:appbar)

**Comment by njovy** [comment]
Thank you for the rapid response and suggestion. I will use NestedScrollView to achieve this.

**Comment by njovy** [comment]
@peng8350 I changed my code to NestedScrollView and it works as expected. One thing I noticed is that the scrolling is not performant with the same code base. Scrolling is very laggy even in a release mode. ```jumpTo``` feature is almost unusuable.
I am pretty sure this is because of the way NestedScrollView works. Would it be hard to add a feature that non zero index can be a RefreshIndicator? I never had this issue with a CustomScrollView.
